src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsActions.ts

```diff
@@ -0,0 +1,253 @@
+		@ILogService logService: ILogService
+	) {
+		super(RunCellAction.ID, new IMultiStateData<CellExecutionState>([
+			{ key: CellExecutionState.Hidden, value: { label: '', className: '', tooltip: '', hideIcon: true } },
+			{ key: CellExecutionState.Stopped, value: { label: '', className: 'action-label notebook-button masked-pseudo start-outline masked-icon', tooltip: localize('runCell', "Run cell"), commandId: 'notebook.command.runactivecell' } },
+			{ key: CellExecutionState.Running, value: { label: '', className: 'action-label codicon notebook-button toolbarIconStop', tooltip: localize('stopCell', "Cancel execution") } },
+			{ key: CellExecutionState.Error, value: { label: '', className: 'toolbarIconRunError', tooltip: localize('errorRunCell', "Error on last run. Click to run again") } },
+		], CellExecutionState.Hidden), keybindingService, logService);
+		this.ensureContextIsUpdated(context);
+	}
+
+	public override run(): Promise<void> {
+		return this.doRun();
+	}
+
+	public async doRun(): Promise<void> {
+		if (!this._context) {
+			// TODO should we error?
+			return;
+		}
+		try {
+			await this._context.cell.runCell(this.notificationService, this.connectionManagementService);
+		} catch (error) {
+			let message = getErrorMessage(error);
+			this.notificationService.error(message);
+		}
+	}
+
+	private ensureContextIsUpdated(context: CellContext) {
+		if (context && context !== this._context) {
+			if (this._executionChangedDisposable) {
+				this._executionChangedDisposable.dispose();
+			}
+			this._context = context;
+			this.updateStateAndExecutionCount(context.cell.executionState);
+			this._executionChangedDisposable = this._context.cell.onExecutionStateChange((state) => {
+				this.updateStateAndExecutionCount(state);
+			});
+		}
+	}
+
+	private updateStateAndExecutionCount(state: CellExecutionState) {
+		let label = '';
+		let className = '';
+		if (!types.isUndefinedOrNull(this._context.cell.executionCount)) {
+			label = `[${this._context.cell.executionCount}]`;
+			// Heuristic to try and align correctly independent of execution count length. Moving left margin
+			// back by a few px seems to make things "work" OK, but isn't a super clean solution
+			if (label.length === 4) {
+				className = 'execCountTen';
+			} else if (label.length > 4) {
+				className = 'execCountHundred';
+			}
+		}
+		this.states.updateStateData(CellExecutionState.Hidden, (data) => {
+			data.label = label;
+			data.className = className;
+		});
+		this.updateState(state);
+	}
+}
+
+export class HideCellAction extends Action {
+	private static readonly ID = 'hideCell';
+	private static readonly LABEL = undefined;
+	private static readonly ICON = 'notebook-button delete masked-icon';
+
+	constructor(
+		private hideFn: () => void,
+		private context: any
+	) {
+		super(HideCellAction.ID, HideCellAction.LABEL, HideCellAction.ICON);
+	}
+
+	override run(): Promise<void> {
+		return new Promise<void>((resolve) => {
```

Comment:
Is this even needed? You're not awaiting on anything so seems like you could just call the function directly

Reply:
VSCode complaints if I don't wrap it in a promise. I think it's because _apply_ does not return a promise, and since the function has to return a promise, I have to wrap it in one.

Reply:
Two options : 

1. Make the function async so it's automatically wrapped in a promise
2. `return Promise.resolve()`

Reply:
Thanks! Option 1 worked great.